cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java

```diff
@@ -539,14 +540,42 @@ protected static void addAppArtifact(Id.Artifact artifactId, Class<?> appClass,
    *
    * @param artifactId the id of the artifact to add
    * @param appClass the application class to build the artifact from
+   * @param exportPackages the packages to export and place in the manifest of the jar to build. This should include
+   *                       packages that contain classes that plugins for the application will implement.
+   * @return an {@link ArtifactManager} to manage the added artifact
+   */
+  protected static ArtifactManager addAppArtifact(NamespacedArtifactId artifactId, Class<?> appClass,
+                                                  String... exportPackages) throws Exception {
+    return getTestManager().addAppArtifact(artifactId, appClass, exportPackages);
+  }
+
+  /**
+   * Build an application artifact from the specified class and then add it.
+   *
+   * @param artifactId the id of the artifact to add
+   * @param appClass the application class to build the artifact from
    * @param manifest the manifest to use when building the jar
-   * @throws Exception
+   * @deprecated since 3.4.0. Use {@link #addAppArtifact(NamespacedArtifactId, Class, Manifest)}
```

Comment:
I think its because of this https://github.com/caskdata/cdap/blob/develop/cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactId.java